apps/passport-server/src/services/generic-issuance/genericIssuanceService.ts

```diff
@@ -241,6 +251,104 @@ export class GenericIssuanceService {
     return relevantCapability.checkin(req);
   }
 
+  public async getAllUserPipelineDefinitions(
+    userId: string
+  ): Promise<PipelineDefinition[]> {
+    // TODO: Add logic for isAdmin = true
+    return (await this.definitionDB.loadPipelineDefinitions()).filter((d) =>
+      this.userHasPipelineDefinitionAccess(userId, d)
+    );
+  }
+
+  private userHasPipelineDefinitionAccess(
+    userId: string,
+    pipeline: PipelineDefinition
+  ): boolean {
+    return (
+      pipeline.ownerUserId === userId || pipeline.editorUserIds.includes(userId)
+    );
+  }
+
+  public async getPipelineDefinition(
+    userId: string,
+    pipelineId: string
+  ): Promise<PipelineDefinition> {
+    const pipeline = await this.definitionDB.getDefinition(pipelineId);
+    if (!pipeline || !this.userHasPipelineDefinitionAccess(userId, pipeline))
+      throw new PCDHTTPError(404, "Pipeline not found or not accessible");
+    return pipeline;
+  }
+
+  public async upsertPipelineDefinition(
```

Comment:
One thing this doesn't do is cause the generic issuance server to actually load the pipeline. It gets stored in `this.definitionDB`, but `this.pipelines` doesn't have the pipeline in it until `this.createPipelines()` is called. Possibly we want something like `reloadPipeline(pipelineId)` for when a pipeline has changed.

Reply:
Discussed over call - will integrate this later today once items like #1452 have been merged

Reply:
good point